lib/membrane_rtmp_plugin/rtmp_server.ex

```diff
@@ -39,7 +40,22 @@ defmodule Membrane.RTMP.Server do
   @spec start_link(server_options :: t()) :: GenServer.on_start()
   def start_link(server_options) do
     gen_server_opts = if server_options[:name] == nil, do: [], else: [name: server_options[:name]]
+
     server_options = Enum.into(server_options, %{})
+
+    server_options =
+      if Map.get(server_options, :new_client_callback, nil) == nil do
```

Comment:
[NIT] I believe you could do as well: 
```suggestion
      if server_options[:new_client_callback] == nil do
```